### ISSUE TITLE
Feature Request: Allow Optional Comparison Branch for Changelog

### ISSUE THREAD

**Issue by SimonHenz97** [AOEpeople/semanticore#71]
#### Current Functionality

From my understanding, Semanticore currently operates as follows:
1. A push to Branch A triggers the pipeline.
2. The pipeline creates a release merge request.
3. This merge request is then merged back into Branch A.
4. The release gets created
5. A new push to A creates a new Merge Request with updated Changelog

#### Desired Workflow

At my company, we have a slightly different workflow:
1. Branch A is the main release branch.
2. Branch B is the development branch.
3. All feature branches are merged into Branch B.
4. Once all tasks are completed, Branch B is merged into Branch A for the release.

#### Problem

Using Semanticore with this workflow, we encounter an issue:
- Semanticore runs on both Branch A and Branch B.
- When enough tasks are completed in Branch B, we merge the Semanticore-generated merge request into Branch A.
- However, subsequent changes in Branch B do not recognize the current release from Branch A, resulting in a new, redundant changelog.

#### Proposed Solution

 **Specify Monitoring Branch in Semanticore:**
   - Introduce a configuration option in Semanticore to specify which branch it should monitor for the current release.
   - This would allow Branch B to recognize the release from Branch A and maintain an accurate changelog.

### Conclusion

Implementing either of these solutions would greatly enhance Semanticore's flexibility and usability in different branching workflows, particularly in cases where a development branch and a main release branch are used.

If there currently exists a way to work like this with Semanticore please let me know, we really want to use this tool!

**Comment by bastianccm** [comment]
Hi @SimonHenz97 

that is a nice idea and should be possible without too much work, I'll try to implement it.
